src/ui/HeaderButton.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Button as ChakraButton } from '@chakra-ui/react';
+import useColorModeValue from './hooks/useColorModeValue';
+
+export type ButtonProps = React.ComponentPropsWithRef<typeof ChakraButton>;
+
+const HeaderButton = React.forwardRef<HTMLButtonElement, ButtonProps>(
+  (props, ref) => {
+    const color = useColorModeValue('gray.700', 'gray.100', 'gray.700');
+    return (
+      <ChakraButton
+        ref={ref}
+        p={1}
+        border="none"
+        bg="transparent"
+        textTransform="uppercase"
+        fontSize={0}
+        color={color}
+        _hover={{ bgColor: 'transparent' }}
+        _active={{ bgColor: 'transparent' }}
+        {...props}
+      />
+    );
```

Comment:
After thinking about it, I know I told you to not use the `headerButton` variant in the theme and make a separate component... buuut I think actually we should just make this the default button style in the theme. So not a variant, just the default (I think the default is a solid variant so this would be the styles for the solid variant). That way any button anywhere adapts to the color mode, which is how it should work I think

Reply:
Settings button can still remain separate because that's a more one-off case

Reply:
Yea, I like the default color-adaptive variant idea.